dids/diddht/serialization.go

```diff
@@ -0,0 +1,90 @@
+package diddht
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/tbd54566975/web5-go/crypto/dsa"
+	"github.com/tbd54566975/web5-go/dids/didcore"
+)
+
+// UnmarshalVerificationMethod unpacks the TXT DNS resource encoded verification method
+func UnmarshalVerificationMethod(data string) (*didcore.VerificationMethod, error) {
+	propertyMap := parseTXTRecordData(data)
+
+	vm := &didcore.VerificationMethod{}
+	var key string
+	var algorithmID string
+	for property, v := range propertyMap {
+		switch property {
+		// According to https://did-dht.com/#verification-methods, this should not be a list
+		case "id":
+			vm.ID = strings.Join(v, "")
+		case "t": // Index of the key type https://did-dht.com/registry/index.html#key-type-index
+			algorithmID, _ = dhtIndexToAlg[strings.Join(v, "")]
+		case "k": // unpadded base64URL representation of the public key
+			key = strings.Join(v, "")
+		case "c": // the controller is optional
+			vm.Controller = strings.Join(v, "")
+		default:
+			continue
+		}
+	}
+
+	if len(key) <= 0 || len(algorithmID) <= 0 {
+		return nil, fmt.Errorf("unable to parse public key")
+	}
+
+	// RawURLEncoding is the same as URLEncoding but omits padding.
+	// Decoding and reencoding to make sure there is no padding
+	keyBytes, err := base64.RawURLEncoding.DecodeString(key)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(keyBytes) <= 0 {
+		return nil, fmt.Errorf("empty key")
+	}
+
+	j, err := dsa.BytesToPublicKey(algorithmID, keyBytes)
+	if err != nil {
+		return nil, err
+	}
+	vm.PublicKeyJwk = &j
+
+	// validate all the parts exist
+	if len(vm.ID) <= 0 || vm.PublicKeyJwk == nil {
+		return nil, fmt.Errorf("malformed verification method representation")
+	}
+
+	return vm, nil
+}
+
+// UnmarshalService unpacks the TXT DNS resource encoded service
+func UnmarshalService(data string) *didcore.Service {
+	propertyMap := parseTXTRecordData(data)
```

Comment:
I'm pretty surprised this function and `parseTXTRecordData()` don't return errors?

Reply:
<img src="https://media3.giphy.com/media/jRN27EUBzMiIdPVPZp/giphy.gif"/>